crates/ruff_linter/src/rules/isort/categorize.rs

```diff
@@ -72,16 +73,22 @@ pub(crate) fn categorize<'a>(
     detect_same_package: bool,
     known_modules: &'a KnownModules,
     target_version: PythonVersion,
+    no_sections: bool,
 ) -> &'a ImportSection {
     let module_base = module_name.split('.').next().unwrap();
     let (import_type, reason) = {
-        if level.is_some_and(|level| level > 0) {
+        if !no_sections && level.is_some_and(|level| level > 0) {
             (
                 &ImportSection::Known(ImportType::LocalFolder),
                 Reason::NonZeroLevel,
             )
         } else if module_base == "__future__" {
             (&ImportSection::Known(ImportType::Future), Reason::Future)
```

Comment:
Does isort put the `__future__` imports in their own section, even if you pass `--no-sections`?